operator/v1/types_network.go

```diff
@@ -533,6 +536,21 @@ type ProxyConfig struct {
 	ProxyArguments map[string]ProxyArgumentList `json:"proxyArguments,omitempty"`
 }
 
+// EgressIPConfig defines the configuration knobs for egressip
+type EgressIPConfig struct {
+	// ReachabilityTotalTimeout configures the EgressIP node reachability check total timeout in seconds.
+	// If the EgressIP node cannot be reached within this timeout, the node is declared down.
+	// Setting a large value may cause the EgressIP feature to react slowly to node changes.
+	// In particular, it may react slowly for EgressIP nodes that really have a genuine problem and are unreachable.
+	// When omitted, this means the user has no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is 1 second.
+	// A value of 0 disables the EgressIP node's reachability check.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=10
```

Comment:
So we are setting a maximum of 10s, can you quickly explain why you chose that value before we merge, just want to make sure that's going to be enough (this is smaller than I thought it would be)

Reply:
I picked this value given osdn platform was shipping with 5s and given its impact if its too large I went with 10s I am checking with @MichaelWasher to confirm

Reply:
Well... I mean it all depends on the rety-count within this time. The concern is not that a single packet takes more than 5s/10s to complete. It's more about how many times is the packets are sent. By default with TCP, the first SYN will retry after 5.8s, so you're only getting 1 retry.

Reply:
However reviewing the new dialer code, it looks like you're retrying ever second-ish (0.9s) so that'll be ~10 retries, which should be a tonne. IMO it LGTM

Reply:
its not meant to be long like kubelet for the side effects mentioned since the default is `1sec` the upper bond shouldn't be that far off IMO hence the 10sec

Reply:
The side effect for this and kubelet, as I understand it, isn't particularly different. We have something that's being healthchecked, we take it down/offline at some point if the healthchecks are failing. This would be similar to pods as well with readyz checks, which are rarely as short as 10s in my experience. I think having such a low limit will confuse users

Reply:
60s will be good upper limit?

Reply:
@JoelSpeed , the difference between this and the Kubelet is that if a Node actually goes down, all traffic will be lost. When this was originally deployed, the timeout was 1s with no retries. 60s is a massive jump from this, however as it's the upper bound, it doesn't really matter.

Reply:
TL;DR: the criteria should be: how did the feature behave before we made this a knob. And that should be the value we pick (or something close to that). Not: how does any unrelated feature in the K8s API normally select timeouts.

We use this as a detection mechanism for EgressIP failover. Imagine EgressIP as a HSRP or VRRP VIP, just the other way around. All traffic to the outside world is funneled through that node from EgressIP marked pods, so potentially with a 60 seconds timeout by default, we cause those pods' traffic to be blackholed  for 60 seconds in case of a node issue. So the shorter the timeout (within a reasonable delay) the better.

IMO, the criteria should be: how did the feature behave before we made this a knob. And that should be the value we pick (or something close to that). Not: how does any unrelated feature in the K8s API normally select timeouts. If your ready probe on a pod takes a minute to time out, then your one pods is affected. If the reachability check for EgressIP takes 60 seconds (or 4 minutes) to timeout, then potentially dozens of pods can't reach the outside world for that time.  Add to this that customers expected this to happen within a second, and now from one version of the product we bump this to a value 60 times higher ... seems a lot to me. And add to this again that node reboots are actually really common in OCP due to the machine config operator taking them offline and online.

Just to clarify: the dialer isn't actually *establishing* TCP connections, right. Instead, it's calling into tcp/9 with no listener on the other side, it's get a port unreachable message or equivalent back, and thus it will know that the host on the other side is up (there would probably have been better ways of detecting that the node is up, but this  is how it was done in SDN and OVNK way before this change). Thus, we don't have to worry about when TCP retries the first time as we'll close the connection ourselves. Regardless if the current detection mechanism is good or not, on the API side, I'd push for a shorter delay rather than a longer delay because of the aforementioned reasons.

https://github.com/ovn-org/ovn-kubernetes/blob/ce8b60f2309dad9aa3473c62c949ecf75ff2d185/go-controller/pkg/ovn/egressip.go#L2167